src/Microsoft.AspNet.Identity/RoleManager.cs

```diff
@@ -162,20 +162,23 @@ public void Dispose()
             {
                 throw new ArgumentNullException("role");
             }
-
             var result = await ValidateRoleInternal(role);
             if (!result.Succeeded)
             {
                 return result;
             }
             await UpdateNormalizedRoleNameAsync(role);
-            return await LogResultAsync(await Store.CreateAsync(role, CancellationToken), role);
+            result = await Store.CreateAsync(role, CancellationToken);
+            using (IdentityLogger.BeginScope(await GetScopeMessageforRoleAsync(role)))
+            {
+                return IdentityLogger.Log(result);
```

Comment:
If we want to get tricky, we could capture the messages and only create the scope if we have more than one, since we have a using block, we have that control (to defer the logging calls until the end of the using and then decide to really create the scope or not...  I'm not sure its worth it though)